### DISCUSSION TITLE
Refresh data on history back

### DISCUSSION THREAD

**Discussion by tb-b** [vercel/next.js#45098] [Help]
What would be the recommended way of dealing with a similar scenario:

- user avatar is shown on main page
- user goes to /account and changes the avatar
- this triggers route.refresh() (or revalidate, but I'm running into auth cookie issue described [here](https://github.com/vercel/next.js/discussions/45094))
- user goes back to the main page (via browser back button), but the avatar is now outdated

**Comment by GabenGar** [comment]
Store avatar in context/redux store and derive the component image from it.
Won't need to do the hacky `route.refresh()` also.
